include/seastar/core/memory.hh

```diff
@@ -382,29 +383,76 @@ public:
     void operator=(scoped_large_allocation_warning_disable&&) = delete;
 };
 
-/// Enable/disable heap profiling.
+/// Describes an allocation location in the code.  The location is identified by
+/// its backtrace. One allocation_site can represent many allocations at the
+/// same location. `count` and `size` represent the cumulative sum of all
```

Comment:
To be clear, count and sum represent the cumulative sum for "some set" of allocations at the call site, depending on how the reset of the code manages sampling, right?

So it in our case it's something like "all sampled allocations which are still lived".

Not to be pedantic but to make clear these values need to be interpreted in the context of how the rest of the code manages samples (which means that e.g., we need to gross-up size values to reflect the full heap value in another tool if we want that, etc).

Reply:
Good point, I will clarify that. Right now it is infact the grossed up value (see sampler.sample_size() and its usage). I.e.: we don't store the original size but the how-much-it-should-count for size directly. So no post processing is needed.

We could obviously store the original size but I currently don't see an advantage of doing that given that making use of that would require additional information (storing per allocation information) and it has the benefit that in postprocessing you don't need the sampling rate.